### PR TITLE
[KEYCLOAK-2052] Allows independently set timeouts for e-mail verification link and rest e.g. forgot password link

### PR DESCRIPTION
**Pull request by abstractj**
After the merge of this PR, please also merge the docs here https://github.com/keycloak/keycloak-documentation/pull/233

### CODE REVIEW

File: testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/realm/TokensTest.java [hunk 1]
```diff
@@ -78,10 +96,94 @@ public void testLifespanOfRealmSession() throws InterruptedException {
...
+        assertTrue("User action token for verify e-mail expected",
+                tokenSettingsPage.form().isOperationEquals(VerifyEmailActionToken.TOKEN_TYPE, TIMEOUT, TimeUnit.DAYS));
+
+        assertTrue("User action token for reset credentials expected",
+                tokenSettingsPage.form().isOperationEquals(ResetCredentialsActionToken.TOKEN_TYPE, TIMEOUT, TimeUnit.HOURS));
+
+        //Verify if values were properly propagated
+        Map<String, Integer> userActionTokens = getUserActionTokens();
+
+        assertTrue("Action Token attributes list should contain 2 items", userActionTokens.size() == 2);
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/realm/TokensTest.java]: Better use
```java
assertThat("Action Token attributes list should contain 2 items", userActionTokens, Matchers.hasSize(2));
```
as it gives more descriptive error messages on assertion failure.

File: testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/realm/TokensTest.java [hunk 2]
```diff
@@ -78,10 +96,94 @@ public void testLifespanOfRealmSession() throws InterruptedException {
...
+        assertEquals("VerifyEmailActionToken expected to be propagated",
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/realm/TokensTest.java]:
```java
assertThat(userActionTokens, Matchers.hasEntry(VerifyEmailActionToken.TOKEN_TYPE), Long.toString(TimeUnit.DAYS.toSeconds(TIMEOUT)));
assertThat(userActionTokens, Matchers.hasEntry(ResetCredentialsActionToken.TOKEN_TYPE), Long.toString(TimeUnit.HOURS.toSeconds(TIMEOUT)));
```
that allows removing the following two asserts.

File: themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties [hunk 3]
```diff
@@ -114,6 +114,15 @@ action-token-generated-by-admin-lifespan=Default Admin-Initiated Action Lifespan
 action-token-generated-by-admin-lifespan.tooltip=Maximum time before an action permit sent to a user by admin is expired. This value is recommended to be long to allow admins send e-mails for users that are currently offline. The default timeout can be overridden right before issuing the token.
 action-token-generated-by-user-lifespan=User-Initiated Action Lifespan
 action-token-generated-by-user-lifespan.tooltip=Maximum time before an action permit sent by a user (e.g. forgot password e-mail) is expired. This value is recommended to be short because it is expected that the user would react to self-created action quickly.
+
+action-token-generated-by-user.execute-actions=Execute Actions
+action-token-generated-by-user.idp-verify-account-via-email=Identity Provider E-mail Verification
```
Comment (hmlnarik) [inline themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties]: Identity Provider *Account* e-mail Verification ?

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thanks for the changes, overall looks good to me, please only fix few nits and squash to single commit.

**Comment by abstractj** [comment]
@hmlnarik I did all the changes you mentioned with some minor changes to what you suggested. I only disagree with a single "nit" :) Define *Identity Provider Account e-mail Verification*, instead of something like *IdP Account E-mail Verification*. But I did the change anyways.

I understand the idea of something user-friendly, at the same time, people are familiar with the term IdP. Also, this is gonna make our UI component huge and not perfectly aligned with other selects. See the screenshot:

**Comment by hmlnarik** [comment]
@abstractj ack, IdP makes sense and it would make the UI cleaner. Go for it :-) And thanks for the changes!

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**